api/src/main/java/org/eclipse/microprofile/config/inject/ConfigProperties.java

```diff
@@ -64,10 +65,36 @@
 @Qualifier
 public @interface ConfigProperties {
 
+
     /**
      * The prefix of the configuration properties
      * @return the configuration property prefix
      */
     @Nonbinding
     String prefix() default "";
+
+    /**
+     * Support inline instantiation of the {@link ConfigProperties} qualifier.
+     */
+    public final static class Literal extends AnnotationLiteral<ConfigProperties> implements ConfigProperties {
+
+        public static final Literal NOPREFIX = of(""); 
+
+        private static final long serialVersionUID = 1L;
+        private final String prefix;
+
+        public static Literal of(String prefix) {
```

Comment:
I don't know what the usual way of doing literals is but I'd make this return `ConfigProperties` as this should be all the user cares about.

Reply:
It needs to return Literal as it is used in CDI.current().select(Class, Annotation...). See [here](https://github.com/eclipse-ee4j/cdi/blob/master/api/src/main/java/jakarta/enterprise/context/Initialized.java) for an example. All of the CDI spec Qualifier has this inline Literal defined.

Reply:
@jbee please see my comments above to see whether it answers your question.

Reply:
@Emily-Jiang it does. I can see this is used as a convention. It is not necessary though to call `CDI.current().select(Class, Annotation...)` as it only expects `Annotation` which the annotation interface still extends.

Reply:
@jbee it is a good usability experience. It provides out of box Annotation implementation rather than writing it yourself.

Reply:
My comment was only on the used return type. It is unnecessary to expose it as `Literal` as the annotation interface, here `ConfigProperties` has all the public methods ever used by users of the `Literal` class. Not really important.

Reply:
@jbee The return type `ConfigProperties` is not relevant though as the main purpose is `Annotation` type, which `AnnotationLiteral` implements. I might have misunerstood what you meant. The static class is about its annotation literal not this annotation itself. Can you demonstrate what you meant via code example and its usage?

Reply:
I think we getting off-topic with this discussion. Its no big deal and the convention seems to be to use the `Literal` class name as return type. I only commented on it as it is usually common practice to no expose implementation classes in factory methods if there is an interface and if that interface is sufficient. As you note it even would be sufficient to use `Annotation` itself.

Reply:
I see. I just used the common pattern, established by CDI. Thanks @jbee!